CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/SplitNodesPlugin.java

```diff
@@ -130,82 +144,132 @@ public void updateParameters(Graph graph, PluginParameters parameters) {
+                    }
+                }
+            }
+        } else {
+            //Add a transaction of the type 'linkType' between the selected node and new node
+            int newTransactionId = graph.addTransaction(selectedNode, newVertexId, directed);
+            graph.setStringValue(transactionTypeAttributeId, newTransactionId, linkType);
         }
-        result.set(GraphRecordStoreUtilities.TRANSACTION + AnalyticConcept.TransactionAttribute.TYPE, linkType);
+        return newVertexId;
     }
 
     @Override
-    protected RecordStore query(final RecordStore query, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException {
-        final RecordStore result = new GraphRecordStore();
-
+    public void edit(final GraphWriteMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException, PluginException {
         final Map<String, PluginParameter<?>> splitParameters = parameters.getParameters();
         final String character = splitParameters.get(SPLIT_PARAMETER_ID) != null && splitParameters.get(SPLIT_PARAMETER_ID).getStringValue() != null ? splitParameters.get(SPLIT_PARAMETER_ID).getStringValue() : "";
         final ParameterValue transactionTypeChoice = splitParameters.get(TRANSACTION_TYPE_PARAMETER_ID).getSingleChoice();
         final String linkType = transactionTypeChoice != null ? transactionTypeChoice.toString() : AnalyticConcept.TransactionType.CORRELATION.getName();
         final boolean allOccurrences = splitParameters.get(ALL_OCCURRENCES_PARAMETER_ID).getBooleanValue();
+        final boolean splitIntoSameLevel = splitParameters.get(SPLIT_INTO_SAME_LEVEL_PARAMETER_ID).getBooleanValue();
+
+        int vertexSelectedAttributeId = graph.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.SELECTED.getName());
+        int vertexIdentifierAttributeId = graph.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.IDENTIFIER.getName());
+        List<Integer> newVertices = new ArrayList<>();
+
+        final int graphVertexCount = graph.getVertexCount();
+        for (int position = 0; position < graphVertexCount; position++) {
+            int currentVertexId = graph.getVertex(position);
+
+            if (graph.getBooleanValue(vertexSelectedAttributeId, currentVertexId)) {
+                final String identifier = graph.getStringValue(vertexIdentifierAttributeId, currentVertexId);
+
+                if (identifier != null && identifier.contains(character) && identifier.indexOf(character) < identifier.length() - character.length()) {
+                    String leftNodeIdentifier = "";
+                    if (allOccurrences) {
+                        final List<String> substrings = new ArrayList<>(Arrays.asList(identifier.split(character)));
```

Comment:
given the use of `substrings`, I would probably just leave it as an array rather than converting it to a list